src/plugins/condition/components/inspector/StylesView.vue

```diff
@@ -0,0 +1,600 @@
+                        subObjects.push(item);
+                        itemStyle = getApplicableStylesForItem(item);
+                        if (this.hasConditionalStyle(item)) {
+                            itemsWithConditionalStyles += 1;
+                        }
+                    } else {
+                        itemStyle = getApplicableStylesForItem(domainObject, layoutItem || item);
+                        this.items.push({
+                            id: layoutItem.id,
+                            applicableStyles: itemStyle
+                        });
+                        if (this.hasConditionalStyle(item, layoutItem)) {
+                            itemsWithConditionalStyles += 1;
+                        }
+                    }
+                }
+                itemInitialStyles.push(itemStyle);
+            });
+            this.isStaticAndConditionalStyles = this.isMultipleSelection && itemsWithConditionalStyles;
+            const {styles, mixedStyles} = getConsolidatedStyleValues(itemInitialStyles);
+            this.initialStyles = styles;
+            this.mixedStyles = mixedStyles;
+
+            this.domainObject = domainObject;
+            this.removeListeners();
+            if (this.domainObject) {
+                this.stopObserving = this.openmct.objects.observe(this.domainObject, '*', newDomainObject => this.domainObject = newDomainObject);
+                this.stopObservingItems = this.openmct.objects.observe(this.domainObject, 'configuration.items', this.updateDomainObjectItemStyles);
+            }
+
+            subObjects.forEach(this.registerListener);
+        },
+        updateDomainObjectItemStyles(newItems) {
+            //check that all items that have been styles still exist. Otherwise delete those styles
+            let keys = Object.keys(this.domainObject.configuration.objectStyles || {});
+            keys.forEach((key) => {
+                if ((key !== 'styles') &&
+                    (key !== 'staticStyle') &&
+                    (key !== 'conditionSetIdentifier')) {
+                    if (!(newItems.find(item => item.id === key))) {
+                        this.removeItemStyles(key);
+                    }
+                }
+            });
+        },
+        registerListener(domainObject) {
+            let id = this.openmct.objects.makeKeyString(domainObject.identifier);
+
+            if (!this.domainObjectsById) {
+                this.domainObjectsById = {};
+            }
+
+            if (!this.domainObjectsById[id]) {
+                this.domainObjectsById[id] = domainObject;
+                this.observeObject(domainObject, id);
+            }
+        },
+        observeObject(domainObject, id) {
+            let unobserveObject = this.openmct.objects.observe(domainObject, '*', function (newObject) {
+                this.domainObjectsById[id] = JSON.parse(JSON.stringify(newObject));
```

Comment:
We can use an arrow function and get rid of the bind call

Reply:
Done.